daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/CommandPreprocessor.scala

```diff
@@ -237,6 +248,32 @@ private[engine] class CommandPreprocessor(compiledPackages: ConcurrentCompiledPa
     exceptionToResultError(go(0, ty0, v0.value))
   }
 
+  private[engine] def labeledRecordToMap(
+      fields: ImmArray[(Option[String], Value[AbsoluteContractId])])
+    : Option[Map[String, Value[AbsoluteContractId]]] = {
+    @tailrec
+    def go(
+        fields: ImmArray[(Option[String], Value[AbsoluteContractId])],
+        map: Map[String, Value[AbsoluteContractId]])
+      : Option[Map[String, Value[AbsoluteContractId]]] = {
+      fields match {
+        case ImmArrayCons(LabeledValue(label, value), tail) =>
+          go(tail, map + (label -> value))
```

Comment:
What's append if the label is repeated ?
I think the desirialization should failed.

Reply:
Very good point, thanks! I'm adding a test and ruling this out, thanks.

Reply:
Added a test, turns out this is already covered: the possible cases are that
* you have all keys and one is repeated, meaning a wrong number of keys, which is already checked before
* you have an equal number of repeated and missing keys, which is still caught as missing keys are reported